packages/job-components/src/job-schemas.ts

```diff
@@ -254,17 +255,44 @@ export function jobSchema(context: Context): convict.Schema<any> {
                 // processor requires port X this code should throw an error.
                 if (arr != null) {
                     if (!Array.isArray(arr)) {
-                        throw new Error('external_ports is required to be an array');
+                        throw new Error('external_ports is required to be an array of numbers or objects like {name: \'myJob\', port: 9000}.');
                     }
-                    if (arr.includes(45680)) {
-                        throw new Error('Port 45680 cannot be included in external_ports, it is reserved by Teraslice.');
+                    for (const portValue of arr) {
+                        if (isNumber(portValue)) {
+                            if (portValue === 45680) {
+                                throw new Error('Port 45680 cannot be included in external_ports, it is reserved by Teraslice.');
+                            }
+                        } else if (isPlainObject(portValue)) {
+                            Object.entries(portValue).forEach(([key, val]) => {
+                                if (key == null || key === '') {
+                                    throw new Error('key must be not empty');
+                                }
+
+                                if (val == null || val === '') {
+                                    throw new Error(`value for key "${key}" must be not empty`);
+                                }
+                                // eslint-disable-next-line no-prototype-builtins
+                                if (portValue.hasOwnProperty('name') && portValue.hasOwnProperty('port')) {
```

Comment:
That eslint rule shouldn't be ignored, I would recommend using `hasOwn` from `@terascope/utils`.

```suggestion
                                if (hasOwn(portValue, 'name') && hasOwn(portValue, 'port')) {
```

Reply:
I've addressed this and bumped the patch version.